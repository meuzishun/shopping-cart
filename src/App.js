import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Header from './components/header/Header';
import Cart from './components/cart/Cart';
import Home from './components/home/Home.js';
import Shop from './components/shop/Shop.js';
import Contact from './components/contact/Contact.js';
import Footer from './components/footer/Footer.js';
import './App.css';

function App() {
  return (
    <div>
      <BrowserRouter>
        <Header />
        <Cart />
        <Routes>
          <Route path='/' element={<Home />} />
          <Route path='/shop' element={<Shop />} />
          <Route path='/contact' element={<Contact />} />
        </Routes>
        <Footer />
      </BrowserRouter>
    </div>
  );
}

export default App;
