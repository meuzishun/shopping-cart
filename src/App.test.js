import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

describe('App component', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
