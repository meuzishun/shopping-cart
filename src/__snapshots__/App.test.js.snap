// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <h1>
        logo
      </h1>
      <nav
        class="nav"
      >
        <a
          href="/"
        >
          home
        </a>
        <a
          href="/shop"
        >
          shop
        </a>
        <a
          href="/contact"
        >
          contact
        </a>
        <button>
          cart (0)
        </button>
      </nav>
    </div>
    <div>
      Home
    </div>
    <footer>
      <p>
        Copyright Â© meuzishun 2022
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
